### PR TITLE
r-v8: new package at 3.4.0

### PR DESCRIPTION
**Pull request by codeandkey**

### CODE REVIEW

File: var/spack/repos/builtin/packages/r-v8/package.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    homepage = "https://github.com/jeroen/v8"
+    url      = "https://cloud.r-project.org/src/contrib/V8_3.4.0.tar.gz"
+    list_url = "https://cloud.r-project.org/src/contrib/Archive/V8"
+
+    version('3.4.0', sha256='f5c8a2a03cc1be9f504f47711a0fcd1b962745139c9fb2a10fbd79c4ae103fbd')
+
+    depends_on('r-curl@1.0:', type=('build', 'run'))
+    depends_on('r-jsonlite@1.0:', type=('build', 'run'))
+    depends_on('r-rcpp', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-v8/package.py]: You can remove this duplicate line.

**Comment by adamjstewart** [comment]
Might be missing a dependency:

> V8 engine version 6+ is needed for modern JS and WASM support. On Debian / Ubuntu install either libv8-dev or libnode-dev, on Fedora use v8-devel. The readme has instructions for installing backports on Ubuntu Xenial and Bionic. It is still possible to build this package against the legacy libv8 version 3.14 branch, but these engines only support traditional JavaScript (ES5).

File: var/spack/repos/builtin/packages/r-v8/package.py (hunk 1 above)
Comment (codeandkey) [reply var/spack/repos/builtin/packages/r-v8/package.py]: Oops, removed.

**Review by adamjstewart — APPROVED** [review]

**Comment by codeandkey** [comment]
I think r-v8 does require V8 installed on the system, which may need to be added in another package.

**Comment by adamjstewart** [comment]
Do you want to merge this PR as is or add a V8 package first? Any idea how difficult V8 is to package? I strongly dislike relying on system deps, but when traversing build systems and programming languages it becomes annoying to require dependencies (like is the case for py-sphinx-rtd-theme and npm). Basically, I would prefer adding the dependency first, but I would be willing to merge and wait until someone complains that they don't have V8 on their system and adds a package.

**Comment by codeandkey** [comment]
It's probably okay to merge this package for now; most distributions have v8 packages available ([libv8-dev](https://packages.debian.org/stretch/libv8-dev) in Debian).

**Pull request merged**